avocado/core/runners/utils/messages.py

```diff
@@ -0,0 +1,212 @@
+    def get(cls, result, fail_reason=None, returncode=None):  # pylint: disable=W0221
+        """Creates finished message with  all necessary information.
+
+        :param result: test result
+        :type result values for the statuses defined in
+                     :class: avocado.core.teststatus.STATUSES
+        :param fail_reason: parameter for brief specification, of the failed
+                            result.
+        :type fail_reason: str
+        :param returncode: exit status of runner
+        :return: finished message
+        :rtype: dict
+        """
+        return super(FinishedMessage, cls).get(result=result,
+                                               fail_reason=fail_reason,
+                                               returncode=returncode)
+
+
+class GenericRunningMessage(GenericMessage):
+    message_status = 'running'
+    message_type = None
+
+    @classmethod
+    def _get_running_message(cls, msg):
+        """Prepare a message dict with necessary information for specific type.
+
+        :param msg: message data. If the message is str, it will be encoded
+                    with utf-8.
+        :type msg: str, bytes
+        :return: message dict which can be send to avocado server
+        :rtype: dict
+        """
+        message = {'type': cls.message_type, 'log': msg}
+        if type(msg) is not bytes:
+            msg = msg.encode('utf-8')
+            message.update({'log': msg, 'encoding': 'utf-8'})
+        return message
+
+    @classmethod
+    def get(cls, msg, **kwargs):  # pylint: disable=W0221
+        """Creates running message with  all necessary information.
+
+        :param msg: log of running message
+        :type msg: str
+        :return: running message
+        :rtype: dict
+        """
+        kwargs.update(cls._get_running_message(msg))
+        return super(GenericRunningMessage, cls).get(**kwargs)
+
+
+class LogMessage(GenericRunningMessage):
+    message_type = 'log'
+
+
+class StdoutMessage(GenericRunningMessage):
+    """Creates stdout message with  all necessary information."""
+    message_type = 'stdout'
+
+
+class StderrMessage(GenericRunningMessage):
+    """Creates stderr message with  all necessary information."""
+    message_type = 'stderr'
+
+
+class WhiteboardMessage(GenericRunningMessage):
+    """Creates whiteboard message with  all necessary information."""
+    message_type = 'whiteboard'
+
+
+class FileMessage(GenericRunningMessage):
+    """Creates file message with  all necessary information."""
+    message_type = 'file'
+
+    @classmethod
+    def get(cls, msg, path):  # pylint: disable=W0221
+        super(FileMessage, cls).get(msg=msg, path=path)
+
+
+def _get_message_by_type(message_type):
+    """Get the message class based on message_type"""
+
+    messages = [LogMessage, StdoutMessage, StderrMessage, WhiteboardMessage]
+    for message in messages:
+        if message_type == message.message_type:
+            return message
```

Comment:
nitpick: maybe here we could use a dict:

```python
spported_types = {'log': LogMessage,
                  'file': FileMessage' # that is missing here on your list
                  'stderr': StderrMessage}
return supported_types[message_type]
```

And because of that, maybe this method would be unnecessary.

Reply:
True, this is a better solution.